### DISCUSSION TITLE
Only strings with 2 or more characters can be sent.

### DISCUSSION THREAD

**Discussion by vanhaodev** [alec1o/Netly#47] [Q&A (Question and Answer)] [answered]
(image: client send to server, server doesn't receive if string length < 2)
![Screenshot 2024-05-16 133914](https://github.com/alec1o/Netly/assets/86046654/239c9ef7-9adf-4e9f-a1de-64a1e6bd816b)
How to send one character string?

client:
```
Console.WriteLine("Message: ");
string message = Console.ReadLine();
client.ToData(message ?? "No message.");
```

server 
```
client.OnData(data =>
{
    Console.WriteLine(NE.GetString(data));
});
```

**Comment by alec1o** [comment]
Wait... I'll check this. Are you using TCP or UDP?

**Reply by vanhaodev** [reply]
Im using tcp.

**Reply by alec1o** [reply]
Yes ma'am have bug!  👏
1 byte isn't be sent :1st_place_medal:

**Reply by vanhaodev** [reply]
yes sir congrat!!

**Reply by alec1o** [reply, marked as answer]
I found the problem and is fixed...
problem from Byter: https://github.com/alec1o/Byter/commit/ef3bfb0e3782bcd7cf6a2096ce4563020b7d1117
but for now don't have way to fix on Netly3.x just be fixed in netly 4 that will use Byter 3 (in development)

**Reply by vanhaodev** [reply]
where to download lastest version (stable), i want use version 4 now.

**Reply by alec1o** [reply]
Version 4 isn't stable... Is in development and have big changes, look this :rightwards_pushing_hand:  (https://github.com/alec1o/Netly/discussions/36#discussion-6204441)

**Reply by vanhaodev** [reply]
so i will use version 3 !
I just use TCP & UDP for my mmorpg

**Reply by alec1o** [reply]
Huh... wait! soon I'll improve to Byter v3
TCP and UDP have so big performance and low cpu usage in Netly 4 :100: 
After update Byter to v3 I'll fixes small tasks and impl Byter 3 in Netly 4 and I'll release for you, because TCP and UDP in version 4 is stable. :goal_net:

**Reply by vanhaodev** [reply]
So i'll waiting for 4, fighting!!!
